cmd/gardener-node-agent/app/bootstrappers/nodeagent_kubeconfig.go

```diff
@@ -0,0 +1,47 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package bootstrappers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/spf13/afero"
+	"k8s.io/client-go/rest"
+
+	"github.com/gardener/gardener/pkg/nodeagent"
+	nodeagentv1alpha1 "github.com/gardener/gardener/pkg/nodeagent/apis/config/v1alpha1"
+)
+
+// NodeAgentKubeconfig is a runnable to create the kubeconfig for the node-agent.
+type NodeAgentKubeconfig struct {
+	Log         logr.Logger
+	FS          afero.Afero
+	Cancel      context.CancelFunc
+	Config      *rest.Config
+	MachineName string
+}
+
+// Start performs creation of the gardener-node-agent kubeconfig.
+func (n *NodeAgentKubeconfig) Start(ctx context.Context) error {
```

Comment:
unit test

Reply:
We would need an integration test for the same reason as [here](https://github.com/gardener/gardener/pull/10781#discussion_r1850592431).
The existing test already covers the functionality except the initial check for the existence of kubeconfig.
I did not want to create redundant tests, that's why I did not create a test here.

Reply:
After I refactored `app.go` the file is gone.